spring-kafka/src/test/java/org/springframework/kafka/listener/adapter/FilteringAdapterTests.java

```diff
@@ -72,4 +75,173 @@ public void testBatchFilterAckDiscard() throws Exception {
+
+		// Then
+		assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
+		verify(ack, never()).acknowledge();
+	}
+
+	@Test
+	public void listener_should_be_invoked_on_emptyList_and_ignoreEmptyBatch_false() throws Exception {
+		// Given :
+		final RecordFilterStrategy<String, String> filter = new RecordFilterStrategy<>() {
+			@Override
+			public boolean filter(ConsumerRecord<String, String> consumerRecord) {
+				return true;
+			}
+
+			@Override
+			public List<ConsumerRecord<String, String>> filterBatch(
+					List<ConsumerRecord<String, String>> consumerRecords) {
+				// SUT
+				return List.of();
+			}
+
+			@Override
+			public boolean ignoreEmptyBatch() {
+				// SUT
+				return false;
+			}
+		};
+
+		final BatchAcknowledgingMessageListener<String, String> listener = mock(BatchAcknowledgingMessageListener.class);
+		final FilteringBatchMessageListenerAdapter<String, String> adapter =
+				new FilteringBatchMessageListenerAdapter<String, String>(listener, filter);
+		final List<ConsumerRecord<String, String>> consumerRecords = new ArrayList<>();
+		final Acknowledgment ack = mock(Acknowledgment.class);
+
+		final CountDownLatch latch = new CountDownLatch(1);
+		willAnswer(i -> {
+			latch.countDown();
+			return null;
+		}).given(listener).onMessage(any(List.class), any(Acknowledgment.class));
+
+		// When :
+		adapter.onMessage(consumerRecords, ack, null);
+
+		// Then
+		assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
+		verify(ack, never()).acknowledge();
+	}
+
+	@Test
+	public void listener_should_be_invoked_on_notEmptyList_and_ignoreEmptyBatch_false() throws Exception {
+		// Given :
+		final RecordFilterStrategy<String, String> filter = new RecordFilterStrategy<>() {
+			@Override
+			public boolean filter(ConsumerRecord<String, String> consumerRecord) {
+
+				return true;
+			}
+
+			@Override
+			public List<ConsumerRecord<String, String>> filterBatch(
+					// SUT
+					List<ConsumerRecord<String, String>> consumerRecords) {
+				return consumerRecords;
+			}
+
+			@Override
+			public boolean ignoreEmptyBatch() {
+				// SUT
+				return false;
+			}
+		};
+
+		final BatchAcknowledgingMessageListener<String, String> listener = mock(BatchAcknowledgingMessageListener.class);
+		final FilteringBatchMessageListenerAdapter<String, String> adapter =
+				new FilteringBatchMessageListenerAdapter<String, String>(listener, filter);
+		final List<ConsumerRecord<String, String>> consumerRecords = List.of(new ConsumerRecord<>("hello-topic", 1, 1, "hello-key", "hello-value"));
+		final Acknowledgment ack = mock(Acknowledgment.class);
+
+		final CountDownLatch latch = new CountDownLatch(1);
+		willAnswer(i -> {
+			latch.countDown();
+			return null;
+		}).given(listener).onMessage(any(List.class), any(Acknowledgment.class));
+
+		// When :
+		adapter.onMessage(consumerRecords, ack, null);
+
+		// Then
+		assertThat(latch.await(10, TimeUnit.SECONDS)).isTrue();
+		verify(ack, never()).acknowledge();
+	}
```

Comment:
Every member in the class, including last method of the last inner class, has to be surrounded with blank lines.

Reply:
I see! i fixed it 🙇‍♂️